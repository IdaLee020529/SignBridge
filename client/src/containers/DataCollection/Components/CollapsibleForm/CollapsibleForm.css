.collapsible-content {
    position: relative;
    top: 50%;
    left: 50%;
    transform: translate(-50%, 0);
    width: 60%;
    max-width: 100%;
    margin-top: 20px;
    margin-bottom: 20px;
    box-shadow: 2px 3px 10px rgba(0, 0, 0, 0.5);
    padding-bottom: 30px;
    transition: background-color 0.3s;
    transition: width 0.2s;
    background-color: white;
}

.collapsible-content.not-opened {
    cursor: pointer;
}

.collapsible-content.not-opened:hover {
    width: 65%;
    background-color: rgb(224, 224, 224);
}

.collapsible-content.opened {
    width: 65%;
    border: 1px solid #a3a3a3;
}

.collapsible-content-header {
    cursor: pointer;
    display: flex;
    justify-content: space-between;
    align-items: center;
    height: 100%;
    padding: 30px 30px 30px 30px;
}


.collapsible-content-header.admin {
    background-color: #ffe79a
}

.collapsible-content-header.signexpert {
    background-color: #9dadff
}

.header-content {
    display: flex;
    align-items: center;
    /* Align items vertically */
    flex-grow: 1;

}

.header-content .number {
    width: 20%;
}

.header-content .dateTime {
    width: 35%;
}

.header-content .status {
    width: 40%;
}

.expand-icon {
    margin-left: 10px;
}

.collapsible-content-background {
    margin: 0 20px;
    margin-top: 10px;
    padding: 10px;

}

.collapsible-content-details {
    height: auto;
    justify-content: space-evenly;
    margin-bottom: 30px;
}

.header-content h2 {
    margin: 0;
    font-size: 25px;
    color: #000000;
    font-family: Gill Sans;
}

.personal-details,
.sentence-details,
.content-left {
    flex: 1;
    margin: 0 10px;
}

.personal-details h2,
.sentence-details h2,
.content-left h2 {
    display: inline-block;
    padding: 12px 5px;
    color: #3a3a3a;
    font-size: 26px;
    margin-top: 15px;
    /* border-radius: 5px;
    background: #fffd7c;
    box-shadow: 2px 5px 10px rgba(0, 0, 0, 0.2); */
}

.personal-details .ant-descriptions {
    margin-bottom: 20px;
}

.personal-details .ant-descriptions-item-label {
    font-size: 23px;
    color: rgb(68, 68, 68) !important;
    box-shadow: 0px 0px 4px rgba(0, 0, 0, 0.2);
}

.personal-details.admin .ant-descriptions-item-label {
    background-color: #FFF1C5 !important;
}

.personal-details.signexpert .ant-descriptions-item-label {
    background-color: #E7F0FD !important;
}

.personal-details .ant-descriptions-item-content {
    font-size: 20px;
    width: 80% !important;
    box-shadow: 0px 0px 4px rgba(0, 0, 0, 0.2);
}

.sentence-details-content p {
    border-radius: 5px;
    padding: 0 10px;
    background-color: rgb(255, 255, 255);
    font-size: 20px;
}

.collapsible-content-footer {
    display: flex;
    justify-content: space-between;
    align-items: flex-end;
    height: 100%;
}

.collapsible-content-footer .content-left {
    flex-grow: 1;
    margin-left: 15px;
    margin-top: -10px;
}

.collapsible-content-footer .content-left .ant-descriptions {
    width: 70%;
}

.collapsible-content-footer .content-left .ant-descriptions-item-label {
    font-size: 23px;
    color: rgb(68, 68, 68) !important;
    width: 32% !important;
    /* box-shadow: 2px 5px 5px rgba(0, 0, 0, 0.2); */
    box-shadow: 0px 0px 4px rgba(0, 0, 0, 0.2);
}

.collapsible-content-footer .content-left.admin .ant-descriptions-item-label {
    background-color: #FFF1C5 !important;
}

.collapsible-content-footer .content-left.signexpert .ant-descriptions-item-label {
    background-color: #E7F0FD !important;
}


.collapsible-content-footer .content-left .ant-descriptions-item-content {
    font-size: 20px;
    width: 32% !important;
    /* box-shadow: 2px 5px 5px rgba(0, 0, 0, 0.2); */
    box-shadow: 0px 0px 4px rgba(0, 0, 0, 0.2);
}

.collapsible-content-footer .buttons-right {
    margin-right: 10px;
    right: 0;
    bottom: 0
}