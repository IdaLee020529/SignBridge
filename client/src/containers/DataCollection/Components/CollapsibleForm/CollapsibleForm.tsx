import React, { useState } from "react";
import "./CollapsibleForm.css";
import { Button } from "../../../../components/Button/Button";
import VideoInput from "../../../../components/VideoInput/VideoInput";

interface CollapsibleContentProps {
  number: string;
  dateTime: string;
  status: string;
  name: string;
  email: string;
  phoneNumber: string;
  text: string;
  videoLink: string;
  avatar: string;
}

const CollapsibleForm: React.FC<CollapsibleContentProps> = ({
  number,
  dateTime,
  status,
  name,
  email,
  phoneNumber,
  text,
  videoLink,
  avatar,
}) => {
  const [isOpen, setIsOpen] = useState<boolean>(false);

  const toggleOpen = () => {
    if (isOpen) {
      setIsOpen(false);
    }
  };

  const closeForm = () => {
    if (!isOpen) {
      setIsOpen(true);
    }
  };

  return (
    <div
      className={`collapsible-content ${isOpen ? "opened" : "not-opened"}`}
      onClick={closeForm}
    >
      <div className="collapsible-content-header" onClick={toggleOpen}>
        <div className="header-content">
          <h2 className="number">No: {number}</h2>
          <h2 className="dateTime">DateTime: {dateTime}</h2>
          <h2 className="status">Status: {status}</h2>
        </div>
        <div className="expand-icon">
          {isOpen ? <span>&#9650;</span> : <span>&#9660;</span>}
        </div>
      </div>
      {isOpen && (
        <div className="collapsible-content-background">
          <div className="collapsible-content-details">
            <div className="personal-details">
              <h2>Personal Details</h2>
              <div className="personal-details-content">
                <p>
                  <strong>Name:</strong> <span>{name}</span>
                </p>
                <p>
                  <strong>Email:</strong> <span>{email}</span>
                </p>
                <p>
                  <strong>Phone Number:</strong> <span>{phoneNumber}</span>
                </p>
              </div>
            </div>
            <div className="separator"></div>
            <div className="sentence-details">
              <h2>Text/Sentence</h2>
              <div className="sentence-details-content">
                <strong>1.</strong> <span>{text}</span>
              </div>
            </div>
          </div>
          <div className="collapsible-content-footer">
            <div className="content-left">
              <div>
                <strong>Demonstration Video: </strong>
                <a href={videoLink} download>
                  {text}.mp4
                </a>
              </div>

              {status === "In Progress" && (
                <div>
                  <strong>Avatar Video: </strong>
                  <span></span>
                </div>
              )}
            </div>

            <div className="buttons-right">
              {status === "New" && (
                <div className="button-container">
                  <Button
                    type="button"
                    onClick={() => {
                      // Handle accept action
                    }}
                    buttonStyle="btn--accept" // Style for accept button
                    buttonSize="btn--large"
                  >
                    Accept
                  </Button>
                </div>
              )}

              {status === "In Progress" && (
                <div className="button-container">
                  <Button
                    type="button"
                    onClick={() => {}}
                    buttonStyle="btn--send"
                    buttonSize="btn--large"
                  >
                    Send
                  </Button>
                </div>
              )}

              {status === "Rejected" && (
                <div className="button-container">
                  <Button
                    type="button"
                    onClick={() => {}}
                    buttonStyle="btn--send"
                    buttonSize="btn--large"
                  >
                    Send
                  </Button>
                </div>
              )}
            </div>
          </div>
        </div>
      )}
    </div>
  );
};

export default CollapsibleForm;
