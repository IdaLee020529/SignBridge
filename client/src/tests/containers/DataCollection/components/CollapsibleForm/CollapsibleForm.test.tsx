//Import necessary libraries -> either "@testing-library/react" or "react-test-renderer"
import { render, fireEvent } from "@testing-library/react";
import CollapsibleForm from "../../../../../containers/DataCollection/components/CollapsibleForm/CollapsibleForm";

describe("Test Collapsible Form", () => {
  beforeAll(() => {
    Object.defineProperty(window, "matchMedia", {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // Deprecated
        removeListener: jest.fn(), // Deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });
  const props = {
    number: "1",
    dateTime: "2023-11-11",
    status: "New",
    name: "Luffy Gear 5 Sun God Nika",
    email: "luffynika@example.com",
    phoneNumber: "123-456-7890",
    text: "anak ku bodoh",
    videoLink: "testing123.mp4",
    avatarLink: "",
  };
  it("should render correctly", () => {
    const { container, getByText } = render(<CollapsibleForm {...props} />);
    const header = getByText(`No: ${props.number}`).closest(
      ".collapsible-content-header"
    );

    if (header) {
      fireEvent.click(header);
    }

    expect(container).toMatchSnapshot();
  });
});
