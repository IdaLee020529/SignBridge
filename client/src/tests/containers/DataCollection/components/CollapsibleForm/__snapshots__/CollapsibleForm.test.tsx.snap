// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Collapsible Form should render correctly 1`] = `
<div>
  <div
    class="collapsible-content opened"
  >
    <div
      class="collapsible-content-header undefined"
    >
      <div
        class="header-content"
      >
        <h2
          class="number"
        >
          No: 
          1
        </h2>
        <h2
          class="dateTime"
        >
          Date & Time: 
          2023-11-11
        </h2>
        <h2
          class="status"
        >
          Status: 
          New
        </h2>
      </div>
      <div
        class="expand-icon"
      >
        <span>
          â–²
        </span>
      </div>
    </div>
    <div
      class="collapsible-content-background"
    >
      <div
        class="collapsible-content-details"
      >
        <div
          class="personal-details undefined"
        >
          <h2>
            PERSONAL DETAILS
          </h2>
          <div
            class="row"
          >
            <div
              class="col-md-6"
            >
              <div
                class="ant-descriptions ant-descriptions-bordered css-dev-only-do-not-override-42nv3w"
              >
                <div
                  class="ant-descriptions-view"
                >
                  <table>
                    <tbody>
                      <tr
                        class="ant-descriptions-row"
                      >
                        <th
                          class="ant-descriptions-item-label"
                          colspan="1"
                        >
                          <span>
                            Name
                          </span>
                        </th>
                        <td
                          class="ant-descriptions-item-content"
                          colspan="1"
                        >
                          <span>
                            <span
                              class="personal-details-info"
                            >
                              Luffy Gear 5 Sun God Nika
                            </span>
                          </span>
                        </td>
                      </tr>
                      <tr
                        class="ant-descriptions-row"
                      >
                        <th
                          class="ant-descriptions-item-label"
                          colspan="1"
                        >
                          <span>
                            Email
                          </span>
                        </th>
                        <td
                          class="ant-descriptions-item-content"
                          colspan="1"
                        >
                          <span>
                            <span
                              class="personal-details-info"
                            >
                              luffynika@example.com
                            </span>
                          </span>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sentence-details"
        >
          <h2>
            TEXT/ SENTENCE
          </h2>
          <div
            class="sentence-details-content"
          >
            <p>
              anak ku bodoh
            </p>
          </div>
        </div>
      </div>
      <div
        class="collapsible-content-footer"
      >
        <div
          class="content-left undefined"
        >
          <h2>
            VIDEO DETAILS
          </h2>
          <div
            class="row"
          >
            <div
              class="col-md-6"
            >
              <div
                class="ant-descriptions ant-descriptions-bordered css-dev-only-do-not-override-42nv3w"
              >
                <div
                  class="ant-descriptions-view"
                >
                  <table>
                    <tbody>
                      <tr
                        class="ant-descriptions-row"
                      >
                        <th
                          class="ant-descriptions-item-label"
                          colspan="1"
                        >
                          <span>
                            Demonstration Video
                          </span>
                        </th>
                        <th
                          class="ant-descriptions-item-label"
                          colspan="2"
                        >
                          <span>
                            Avatar Video
                          </span>
                        </th>
                      </tr>
                      <tr
                        class="ant-descriptions-row"
                      >
                        <td
                          class="ant-descriptions-item-content"
                          colspan="1"
                        >
                          <span>
                            <span
                              class="video-details-info"
                            >
                              testing123.mp4
                            </span>
                          </span>
                        </td>
                        <td
                          class="ant-descriptions-item-content"
                          colspan="2"
                        >
                          <span>
                            <span
                              class="video-details-info"
                            />
                          </span>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="buttons-right"
        >
          <div
            class="button-container"
          >
            <button
              class="btn btn--accept btn--large"
              type="button"
            >
              Accept
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
